components/eam/src/physics/crm/crm_physics.F90

```diff
@@ -817,40 +913,33 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out, &
       crm_input%vl_esmt(1:ncol,1:pver) = state%v(1:ncol,1:pver)
 #endif /* MMF_ESMT */
+
+      ! Set surface flux variables
+      if (phys_do_flux_avg()) then
+         call pbuf_get_field(pbuf, pbuf_get_index('LHFLX'), shf_ptr)
+         call pbuf_get_field(pbuf, pbuf_get_index('SHFLX'), lhf_ptr)
+         call pbuf_get_field(pbuf, pbuf_get_index('TAUX'),  wsx_ptr)
+         call pbuf_get_field(pbuf, pbuf_get_index('TAUY'),  wsy_ptr)
+         shf_tmp = shf_ptr
+         lhf_tmp = lhf_ptr
+         wsx_tmp = wsx_ptr
+         wsy_tmp = wsy_ptr
+      else
+         shf_tmp = cam_in%shf
+         lhf_tmp = cam_in%lhf
+         wsx_tmp = cam_in%wsx
+         wsy_tmp = cam_in%wsy
+      end if
       do i = 1,ncol
-#ifdef MAML
-         tau00_avg =0._r8
-         bflxls_avg =0._r8
-         fluxu00_avg =0._r8
-         fluxv00_avg =0._r8
-         fluxt00_avg =0._r8
-         fluxq00_avg =0._r8
-         do ii = 1, crm_nx*crm_ny
-            !seems none of variables below is used, so I don't use the CRM-level
-            !input. Later on, they can be changed if needed   
-            tau00_avg = tau00_avg + sqrt(cam_in%wsx(i,ii)**2 + cam_in%wsy(i,ii)**2)
-            bflxls_avg = bflxls_avg + cam_in%shf(i,ii)/cpair + 0.61*state%t(i,pver)*cam_in%lhf(i,ii)/latvap
-            fluxu00_avg = fluxu00_avg + cam_in%wsx(i,ii)          ! N/m2
-            fluxv00_avg = fluxv00_avg + cam_in%wsy(i,ii)          ! N/m2
-            fluxt00_avg = fluxt00_avg + cam_in%shf(i,ii)/cpair    ! K Kg/ (m2 s)
-            fluxq00_avg = fluxq00_avg + cam_in%lhf(i,ii)/latvap   ! Kg/(m2 s)
-         end do
-         crm_input%tau00(i) = tau00_avg*factor_xy 
-         crm_input%bflxls(i) = bflxls_avg*factor_xy 
-         crm_input%fluxu00(i) = fluxu00_avg*factor_xy    ! N/m2
-         crm_input%fluxv00(i) = fluxv00_avg*factor_xy    ! N/m2
-         crm_input%fluxt00(i) = fluxt00_avg*factor_xy    ! K Kg/ (m2 s)
-         crm_input%fluxq00(i) = fluxq00_avg*factor_xy    ! Kg/(m2 s)
-#else
-         crm_input%tau00(i) = sqrt(cam_in%wsx(i)**2 + cam_in%wsy(i)**2)
-         crm_input%bflxls(i) = cam_in%shf(i)/cpair + 0.61*state%t(i,pver)*cam_in%lhf(i)/latvap
-         crm_input%fluxu00(i) = cam_in%wsx(i)     !N/m2
-         crm_input%fluxv00(i) = cam_in%wsy(i)     !N/m2
-         crm_input%fluxt00(i) = cam_in%shf(i)/cpair  ! K Kg/ (m2 s)
-         crm_input%fluxq00(i) = cam_in%lhf(i)/latvap ! Kg/(m2 s)
-#endif  
-         crm_input%wndls(i) = sqrt(state%u(i,pver)**2 + state%v(i,pver)**2)
+         crm_input%tau00(i)   = sqrt(wsx_tmp(i)**2 + wsy_tmp(i)**2)
+         crm_input%bflxls(i)  = shf_tmp(i)/cpair + 0.61*state%t(i,pver)*lhf_tmp(i)/latvap
+         crm_input%fluxu00(i) = wsx_tmp(i)         ! N/m2
+         crm_input%fluxv00(i) = wsy_tmp(i)         ! N/m2
+         crm_input%fluxt00(i) = shf_tmp(i)/cpair   ! K Kg/ (m2 s)
+         crm_input%fluxq00(i) = lhf_tmp(i)/latvap  ! Kg/(m2 s)
+         crm_input%wndls(i)   = sqrt(state%u(i,pver)**2 + state%v(i,pver)**2)
```

Comment:
this is great that we can now apply the smoothed surface fluxes directly to the CRM. I would like to revisit the implication of using the un-smoothed wind speed later. Surface buoyancy and the surface wind speed are used for diagnosing u* (friction vel) and z0(roughnesslength). I know that they are not currently being used in MMF so nothing to be done now.  Just a thought.

Reply:
noted.